### ISSUE TITLE
Ruby dig issue on docker image

### ISSUE THREAD

**Issue by ThatDeveloper** [vitobotta/hetzner-k3s#53]
When running docker container, the following error occurs:

```
Please specify a correct path for the config file.
Traceback (most recent call last):
        8: from /usr/local/bundle/bin/hetzner-k3s:23:in `<main>'
        7: from /usr/local/bundle/bin/hetzner-k3s:23:in `load'
        6: from /usr/local/bundle/gems/hetzner-k3s-0.4.8/exe/hetzner-k3s:4:in `<top (required)>'
        5: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
        4: from /usr/local/bundle/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
        3: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
        2: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
        1: from /usr/local/bundle/gems/hetzner-k3s-0.4.8/lib/hetzner/k3s/cli.rb:28:in `create_cluster'
/usr/local/bundle/gems/hetzner-k3s-0.4.8/lib/hetzner/k3s/cli.rb:355:in `find_hetzner_token': undefined method `dig' for nil:NilClass (NoMethodError)
```

Config file is specified, tried it with various ways. Google tells me this: https://stackoverflow.com/questions/34733913/how-to-avoid-undefined-method-for-nilnilclass-when-working-with-nested-r

Could you please have a look? Thank you very much.

**Comment by ThatDeveloper** [comment]
Verification that it is not a config-file issue:
Even if I use -e on the docker command to supply the token direcly, I get the dig error.
```
Please specify a correct path for the config file.
Traceback (most recent call last):
        8: from /usr/local/bundle/bin/hetzner-k3s:23:in `<main>'
        7: from /usr/local/bundle/bin/hetzner-k3s:23:in `load'
        6: from /usr/local/bundle/gems/hetzner-k3s-0.4.8/exe/hetzner-k3s:4:in `<top (required)>'
        5: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
        4: from /usr/local/bundle/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
        3: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
        2: from /usr/local/bundle/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
        1: from /usr/local/bundle/gems/hetzner-k3s-0.4.8/lib/hetzner/k3s/cli.rb:28:in `create_cluster'
/usr/local/bundle/gems/hetzner-k3s-0.4.8/lib/hetzner/k3s/cluster.rb:27:in `create': undefined method `dig' for nil:NilClass (NoMethodError)
```

**Comment by vitobotta** [comment]
With Docker, did you specify the path of the kubeconfig as `/cluster/kubeconfig` as described in the README?

**Comment by ThatDeveloper** [comment]
Yes

**Comment by vitobotta** [comment]
Can you share the full config file minus the token?

**Comment by ThatDeveloper** [comment]
Without env:
```
docker run --rm -it -v ${PWD}:/cluster -v ${HOME}/.ssh:/tmp/.ssh vitobotta/hetzner-k3s:v0.4.8 create-cluster --config-file config.yaml
```

With env:
```
docker run --rm -it -e HCLOUD_TOKEN='<redacted>' -v ${PWD}:/cluster -v ${HOME}/.ssh:/tmp/.ssh vitobotta/hetzner-k3s:v0.4.8 create-cluster --config-file config.yaml
```

I also tested it with --config-file /cluster/config.yaml and with --config_file.
Different directories were also tried.


```
---
hetzner_token: <redacted>
cluster_name: <redacted>
kubeconfig_path: "/cluster/kubeconfig"
k3s_version: v1.23.1+k3s1
public_ssh_key_path: "~/.ssh/id_rsa.pub"
private_ssh_key_path: "~/.ssh/id_rsa"
ssh_allowed_networks:
  - 0.0.0.0/0
verify_host_key: false
location: nbg1
schedule_workloads_on_masters: false
masters:
  instance_type: cpx11
  instance_count: 3
worker_node_pools:
- name: default
  instance_type: cx41
  instance_count: 2
```

**Comment by vitobotta** [comment]
I've released 0.4.9 and then 0.5.0 with some updates, including a fix for a missing validation when the config file path is invalid. In your case I think the problem is that you specify the config file path as `--config-file config.yaml` while it should be `--config-file /cluster/config.yaml` when you use Docker. I'll check if I need to update the README about it. Please let me know if you still have issues with the new version and with the fixed config file path.
